### PR TITLE
Rework entity mapping for user board relation and user board role

### PR DESCRIPTION
**Pull request by ChernykhVyacheslav**
Reworked entity mapping for user board relation and user board role to be bidirectional. Moved UserBoardRole logic to separate UserBoarRoleService class(was in UserBoardRelationService), added methods to retrieve UserBoardRelation by user id and board id and to get UserBoardRole by UserBoardRelation id.

https://trello.com/c/j9vq6Opw/194-rework-entity-mapping-for-userboardrelation-and-userboardrole

### CODE REVIEW

File: controller/src/main/java/com/softserve/borda/controllers/BoardController.java [hunk 1]
```diff
@@ -462,16 +464,19 @@ public Board getBoard(@PathVariable Long id) {
                                                           @PathVariable Long boardId,
                                                           @PathVariable Long userBoardRoleId) {
         var user = userService.getUserByUsername(authentication.getName());
-        Invitation invitation = new Invitation();
         User receiver = userService.getUserByUsername(receiverUsername);
+        if(user.getId().equals(receiver.getId())) {
+            return  ResponseEntity.badRequest().build();
```
Comment (achehrynets) [inline controller/src/main/java/com/softserve/borda/controllers/BoardController.java]: Add a message about why it is a bad request.

File: shared/src/main/java/com/softserve/borda/config/ModelMapperConfig.java [hunk 2]
```diff
@@ -44,4 +46,16 @@ protected InvitationDTO convert(Invitation invitation) {
             return invitationDTO;
         }
     };
+
+    Converter<UserBoardRelation, UserBoardRelationDTO> userBoardRelationToUserBoardRelationDTO = new AbstractConverter<>() {
+        protected UserBoardRelationDTO convert(UserBoardRelation userBoardRelation) {
```
Comment (achehrynets) [inline shared/src/main/java/com/softserve/borda/config/ModelMapperConfig.java:51]: I think it should have private access.

File: service/src/main/java/com/softserve/borda/services/impl/UserBoardRoleServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                getUserBoardRoleById(userBoardRole.getId());
+        existingUserBoardRole.setName(userBoardRole.getName());
+        return userBoardRoleRepository.save(existingUserBoardRole);
+    }
+
+    @Override
+    public boolean deleteUserBoardRoleById(Long id) {
+        try {
+            userBoardRoleRepository.deleteById(id);
+            return true;
```
Comment (achehrynets) [inline service/src/main/java/com/softserve/borda/services/impl/UserBoardRoleServiceImpl.java:55]: Good practice says to use 1 return in a method.

File: service/pom.xml [hunk 4]
```diff
@@ -24,7 +25,7 @@
     <parent>
         <groupId>com.softserve.borda</groupId>
         <artifactId>borda</artifactId>
-        <version>1.0-SNAPSHOT</version>
+        <version>2.0</version>
```
Comment (achehrynets) [inline service/pom.xml:28]: Should not be ${project.version} value here?

File: persistence/src/main/java/com/softserve/borda/repositories/UserBoardRoleRepository.java [hunk 5]
```diff
@@ -9,4 +9,5 @@
 @Repository
 public interface UserBoardRoleRepository extends JpaRepository<UserBoardRole, Long> {
     Optional<UserBoardRole> findByName(String name);
+    UserBoardRole findByUserBoardRelationsId(Long userBoardRelationId);
```
Comment (achehrynets) [inline persistence/src/main/java/com/softserve/borda/repositories/UserBoardRoleRepository.java]: Should not it return Optional<UserBoardRole>?

**Review by achehrynets — APPROVED** [review]
@ChernykhVyacheslav, In general, looks good.
Left some questions, go through them, and decide if they are applicable.

File: service/pom.xml (hunk 4 above)
Comment (ChernykhVyacheslav) [reply service/pom.xml:28]: It does not allow variables in the parent version

File: persistence/src/main/java/com/softserve/borda/repositories/UserBoardRoleRepository.java (hunk 5 above)
Comment (ChernykhVyacheslav) [reply persistence/src/main/java/com/softserve/borda/repositories/UserBoardRoleRepository.java]: Yes, you are right, fixed it.

File: shared/src/main/java/com/softserve/borda/config/ModelMapperConfig.java (hunk 2 above)
Comment (ChernykhVyacheslav) [reply shared/src/main/java/com/softserve/borda/config/ModelMapperConfig.java:51]: Yes, my bad, fixed it.

File: controller/src/main/java/com/softserve/borda/controllers/BoardController.java (hunk 1 above)
Comment (ChernykhVyacheslav) [reply controller/src/main/java/com/softserve/borda/controllers/BoardController.java]: Fixed

File: service/src/main/java/com/softserve/borda/services/impl/UserBoardRoleServiceImpl.java (hunk 3 above)
Comment (ChernykhVyacheslav) [reply service/src/main/java/com/softserve/borda/services/impl/UserBoardRoleServiceImpl.java:55]: Fixed here and in other services.

File: service/pom.xml (hunk 4 above)
Comment (achehrynets) [reply service/pom.xml:28]: got it

**Pull request merged**